graphql/graphiql
# Monaco Mode - Phase 1

For the first phase, we will only support completion and maybe diagnostics. Summarized more completely in #1445

**Note**: I will be pulling all the LSP changes out of this PR and creating seperate PRs to introduce those, so that this PR will soon be ~20 new files, a few changes and thats it, ideally.

Everything that isn't complete is partially underway. The only issue now is that monaco can't find the AMD id of our custom language worker.

# Outcomes
- introduces `packages/monaco-graphql` and `examples/monaco-webpack`
- Diagnostics (working), Completion Support (todo)
- Required baseline: schema loading, primitive config loading

# Todo
- [x] GraphQL Mode & contribution
- [x] GraphQL Worker Manager
- [ ] GraphQL Worker - written but worker manager will not load it
- [ ] GraphQL Language features
  - [ ] completion - mode providers work, worker doesnt
  - [ ] Diagnostics - mode providers work, worker doesnt
- [ ] Create graphql-config loaders for browser config load by default
- [ ] programatically provide graphql-config ala #1272?
- [ ] handle schema loading/changing schema via monaco service

Scoped to *only* completion for now. Other features will be iterative

# How to Develop/Test
1. check out this branch
2. from the root: `yarn --force`
3. from the root: (watcher) `yarn tsc --watch` (one time) `yarn build-ts`
4. in another terminal tab, from `examples/monaco-graphql`: `yarn start`
